qualtran/_infra/data_types.py

```diff
@@ -899,6 +904,7 @@ class QGF(QDType):
 
     characteristic: SymbolicInt
     degree: SymbolicInt
+    galois_field: Optional['galois.FieldArrayMeta'] = None
```

Comment:
This type is not hashable, as discussed offline please consider updating the class attributes so you only store hashable types and construct the corresponding galois field object by passing in the values to the constructor.

Reply:
@NoureldinYosri Can you please confirm if this was addressed ?

Reply:
QGF was made hashable ... MultiplyPolyByConstantMod is hashable because I overload its __hash__ method ... or do you want to do the same for MultiplyPolyByConstantMod?